Sources/StripeKit/Shared Models/StripeExpandable.swift

```diff
@@ -166,66 +166,66 @@ public struct DynamicExpandable<A: StripeModel, B: StripeModel>: StripeModel {
-//public struct ExpandableCollection<[Model]>: StripeModel {
-//    private enum ExpandableState {
-//        case unexpanded([String])
-//        indirect case expanded([StripeModel])
-//        case empty
-//    }
-//
-//    public init() {
-//        self._state = .empty
-//    }
-//    
-//    public init(from decoder: Decoder) throws {
-//        let codingPath = decoder.codingPath
-//        do {
-//            var container = try decoder.unkeyedContainer()
-//            do {
-//                if try container.decodeNil() {
-//                    _state = .empty
-//                } else {
-//                    _state = .unexpanded(try container.decode([String].self))
-//                }
-//            } catch DecodingError.typeMismatch(let type, _) where type is [String].Type {
-//                _state = .expanded(try container.decode([Model].self))
-//            }
-//        } catch DecodingError.keyNotFound(_, let context) where context.codingPath.count == codingPath.count {
-//            _state = .empty
-//        }
-//    }
-//    
-//    private var _state: ExpandableState
-//    
-//    public func encode(to encoder: Encoder) throws {
-//        var container = encoder.unkeyedContainer()
-//
-//        switch _state {
-//        case let .unexpanded(ids):
-//            try container.encode(ids)
-//        case let .expanded(models):
-//            try container.encode(models)
-//        default:
-//            try container.encodeNil()
-//        }
-//    }
-//    
-//    public var wrappedValue: [String]? {
-//        switch _state {
-//        case .unexpanded(let ids):
-//            return ids
-//        case .expanded(_), .empty:
-//            return nil
-//        }
-//    }
-//        
-//    public var projectedValue: [Model]? {
-//        switch _state {
-//        case .unexpanded(_), .empty:
-//            return nil
-//        case .expanded(let models):
-//            return models
-//        }
-//    }
-//}
+@propertyWrapper
+public struct ExpandableCollection<Model: StripeModel>: StripeModel {
+    private enum ExpandableState {
+        case unexpanded([String])
+        indirect case expanded([Model])
+        case empty
+    }
+
+    public init() {
+        self._state = .empty
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let codingPath = decoder.codingPath
+        do {
+            let container = try decoder.singleValueContainer()
+            do {
+                if container.decodeNil() {
+                    _state = .empty
+                } else {
+                    _state = .unexpanded(try container.decode([String].self))
+                }
+            } catch DecodingError.typeMismatch(let type, _) where type is String.Type {
+                _state = .expanded(try container.decode([Model].self))
+            }
+        } catch DecodingError.keyNotFound(_, let context) where context.codingPath.count == codingPath.count {
+            _state = .empty
+        }
```

Comment:
Define this extension somewhere (in this file if you like), or elsewhere:

```swift
internal extension Decoder {
    func singleValueContainerIfPresentAndNotNull() throws -> SingleValueDecodingContainer? {
        do {
            let container = try self.singleValueContainer()
            
            if container.decodeNil() {
                return nil
            }
            return container
        }
        catch DecodingError.keyNotFound(_, let context)
            where context.codingPath.count == self.codingPath.count
        {
            return nil
        }
    }
}
```
(n.b.: I tinkered with the code formatting for the `catch` part about a dozen times, I just can't figure out something that feels idiomatic for that 😅)

Then you can do this and avoid some of the convoluted boilerplate:

```suggestion
        if let container = try decoder.singleValueContainerIfPresentAndNotNull() {
            do {
                self._state = .unexpanded(try container.decode([String].self))
            } catch DecodingError.typeMismatch(let type, _) where type is String.Type {
                self._state = .expanded(try container.decode([Model].self))
            }
        } else {
            self._state = .empty
        }
```

Obviously you can do the same in `@Expandable` as well.

(Disclaimer: There are no functional changes in this code versus your version; it's entirely a stylistic choice. I personally don't like nested `do` blocks, especially when the errors being explicitly caught are as complex as `DecodingError`s usually are. And of course there's always something to appreciate in factoring out reused logic. `Codable`'s API makes things hard enough to read already.)